lib/cartopy/mpl/geoaxes.py

```diff
@@ -1112,6 +1112,12 @@ def imshow(self, img, *args, **kwargs):
                              'imshow. To hold the data and view limits see '
                              'GeoAxes.hold_limits.')
 
+        if 'origin' not in kwargs:
+            import warnings
+            warnings.warn('The default value for the origin argument will'
+                          ' change from "lower" to "upper" in CartoPy 0.18.',
+                          FutureWarning)
```

Comment:
peanut gallery two cents here, I wonder if the message should explain why the user is seeing this message and what the user should do to avoid seeing the message in the future.  Something like:  "The origin argument was not specified to imshow() and the default value "lower" will change in CartoPy 0.18 to "upper".  Please explicitly set the origin value to ensure forward compatible behavior."

Eh, that is awful wordy, take it or leave it :)